nixos/modules/services/desktops/gnome/gcr-ssh-agent.nix

```
@@ -0,0 +1,52 @@
+{
+  config,
+  options,
+  pkgs,
+  lib,
+  ...
+}:
+let
+  cfg = config.services.gnome.gcr-ssh-agent;
+  opts = options.services.gnome.gcr-ssh-agent;
+  sshCfg = config.programs.ssh;
+  sshOpts = options.programs.ssh;
+in
+{
+  meta = {
+    maintainers = lib.teams.gnome.members;
+  };
+
+  options = {
+    services.gnome.gcr-ssh-agent = {
+      enable = lib.mkOption {
+        default = config.services.gnome.gnome-keyring.enable;
+        defaultText = lib.literalExpression "config.services.gnome.gnome-keyring.enable";
+        example = true;
+        description = "Whether to enable GCR ssh agent.";
+        type = lib.types.bool;
+      };
+
+      package = lib.mkPackageOption pkgs "GCR" {
+        default = [ "gcr_4" ];
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    assertions = lib.singleton {
+      assertion = cfg.enable && sshCfg.startAgent;
```

Since these definitions are already guarded by `mkIf cfg.enable`, you don't need to check `cfg.enable` again here :slightly_smiling_face: 
```suggestion
      assertion = sshCfg.startAgent;
```

---

Oh, actually this is inverted. We want to assert that `startAgent` is _false_ when this module is enabled, so it should be:
```suggestion
      assertion = !sshCfg.startAgent;
```